packages/gatsby-link/src/index.tsx

```diff
@@ -110,7 +133,7 @@ class GatsbyLink extends React.Component {
       // If IO supported and element reference found, setup Observer functionality
       this.io = createIntersectionObserver(ref, inViewPort => {
         if (inViewPort) {
-          this.abortPrefetch = this._prefetch()
+          this.abortPrefetch = this._prefetch() as { abort: () => void } | null
```

Comment:
Type-casting is a code smell, it seems like `abortPrefetch` is not properly typed then